dotcom-rendering/src/components/Discussion.tsx

```diff
@@ -112,15 +112,21 @@ export const Discussion = ({
 	useEffect(() => {
 		setLoading(true);
 		void getDiscussion(shortUrlId, { ...filters, page: commentPage })
-			.then((json) => {
-				setLoading(false);
-				if (json && json.status !== 'error') {
-					setComments(json.discussion.comments);
-					setIsClosedForComments(json.discussion.isClosedForComments);
+			.then((result) => {
+				if (result.kind === 'error') {
+					console.error(`getDiscussion - error: ${result.error}`);
+					return;
 				}
-				if (json?.pages != null) setTotalPages(json.pages);
+
+				setLoading(false);
+				const { pages, discussion } = result.value;
+				setComments(discussion.comments);
+				setIsClosedForComments(discussion.isClosedForComments);
+				setTotalPages(pages);
 			})
-			.catch((e) => console.error(`getDiscussion - error: ${String(e)}`));
+			.catch(() => {
+				// do nothing
```

Comment:
I wonder if there's a way we can still allow errors to be reported? Our `getDiscussion` code could still crash, or the `then` callback here could crash, and this would suppress either of those errors?

Reply:
Good point, we should use `window.guardian.sentry.reportError`.

This could drastically increase the number of errors, though, so we’d want to do this in a separate PR